test/alternator/test_provisioned_throughput.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# Test for ProvisionedThroughput
+# ProvisionedThroughput is part of a table definition
+# The following tests make sure we can get, set and update its value
+
+import pytest
+from botocore.exceptions import ClientError
+from decimal import Decimal
+from test.alternator.util import random_string, random_bytes, new_test_table
+
+# When creating a table with PROVISIONED billing mode, ProvisionedThroughput must be explicitly set,
+# and the same values should be reflected when the table is described.
+# This test ensures that when a table is created with the specified billing mode and ProvisionedThroughput values,
+# the `describeTable` operation returns the correct ProvisionedThroughput values.
+def test_create_table(dynamodb):
+    KeySchema=[ { 'AttributeName': 'p123', 'KeyType': 'HASH' },
+                    { 'AttributeName': 'c4567', 'KeyType': 'RANGE' }
+        ]
+    AttributeDefinitions=[
+                { 'AttributeName': 'p123', 'AttributeType': 'S' },
+                { 'AttributeName': 'c4567', 'AttributeType': 'S' },
+    ]
+
+    ProvisionedThroughput={
+        'ReadCapacityUnits': 6,
+        'WriteCapacityUnits': 7
+    }
+    with new_test_table(dynamodb,
+                        KeySchema=KeySchema,
+        AttributeDefinitions=AttributeDefinitions,
+        BillingMode='PROVISIONED',
+        ProvisionedThroughput=ProvisionedThroughput) as table:
+        got = table.meta.client.describe_table(TableName=table.name)['Table']
+        if 'BillingMode' in got:
+            # PROVISIONED BillingMode can be omitted, only check if it's present
+            assert got['BillingMode'] == 'PROVISIONED'
+        assert got['ProvisionedThroughput']['ReadCapacityUnits'] == ProvisionedThroughput['ReadCapacityUnits']
+        assert got['ProvisionedThroughput']['WriteCapacityUnits'] == ProvisionedThroughput['WriteCapacityUnits']
+
+# When creating a table with PROVISIONED billing mode, ProvisionedThroughput must be explicitly set,
+# and both Read and Write capacity should be present.
+# This test ensures that when a table is created with the specified billing mode both values
+def test_create_table_missing_units(dynamodb):
+    KeySchema=[ { 'AttributeName': 'p123', 'KeyType': 'HASH' },
+                    { 'AttributeName': 'c4567', 'KeyType': 'RANGE' }
+        ]
+    AttributeDefinitions=[
+                { 'AttributeName': 'p123', 'AttributeType': 'S' },
+                { 'AttributeName': 'c4567', 'AttributeType': 'S' },
+    ]
+
+    ProvisionedThroughput={
+        'WriteCapacityUnits': 5
+    }
```

Comment:
a small suggestion (not critical: you can do something like
```python
for ProvisionedThroughput in [{'WriteCapacityUnits': 1}, {'ReadCapacityUnits': 5}]:
    with pytest.raises(...):
      ....
```

P.S. I would use the number 1 when possible, hopefully it will make the test cheaper to run on AWS :-)

Reply:
I wanted two different numbers (making sure we placed the right number at the right place) and wanted that is not zero or one, to make sure it's not some default.

Reply:
Ok.

By the way, please avoid "resolving" the comments that other people made for you - they want to read your answer before the thread disappears! The person who asked the question should resolve it if the answer is satisfactory. Thanks.